### PR TITLE
sycl: fix soft_max_f32 max reduction

### PR DESCRIPTION
**Pull request by someoneinjd**
## Overview

Fix the second-stage max reduction in the SYCL softmax kernel when a workgroup contains more subgroups than `WARP_SIZE`.

Previously, after each subgroup wrote its partial max to `buf_iw[warp_id]`, the cross-subgroup reduction only loaded:

```cpp
max_val = buf_iw[lane_id];
```

This only reduces the first `WARP_SIZE` partial maxima. For example, on Intel GPUs, `WARP_SIZE` can be 16 while the softmax workgroup size is 1024, producing nwarps = 64. In that case, partial maxima from `buf_iw[16..63]` were ignored.
If the true row max is in one of the ignored subgroups, softmax subtracts a too-small max value. For large attention logits this can overflow `exp(x - max)` to `inf`, eventually producing `NaN`.

## Requirements

- I have read and agree with the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md)
- AI usage disclosure: NO

### CODE REVIEW

**Review by arthw — APPROVED** [review]
It's good job!

Thank you!

**Comment by jlionhan** [comment]
This PR also fixes a bug in  VAE processing of stable-diffusion.cpp that caused fully white image to be generated.

Thank you for your contribution!

**Comment by someoneinjd** [comment]
> This PR also fixes a bug in VAE processing of stable-diffusion.cpp that caused fully white image to be generated.
> 
> Thank you for your contribution!

Haha, I actually discovered this problem when running `stable-diffusion.cpp` on my Lunar Lake laptop as well. Glad this could help. 

Once this PR gets merged into `ggml`, I will submit an issue to `stable-diffusion.cpp` requesting them to update `ggml`.

**Review by arthw — APPROVED** [review]
